### PR TITLE
Added the ability for spells to have the attribute to use_stealth.

### PR DESCRIPTION
**Pull request by jayRyan24**
I defaulted Vivi to have this be set to true.

### CODE REVIEW

**Comment by jayRyan24** [comment]
I could use some necros testing this. I will test it tonight with Ssarek.

I could also use some feedback on whether I should keep it as is or change to check if true.

`game_state.hide_on_cast = true if data['use_stealth']` vs `game_state.hide_on_cast = true if data['use_stealth'] == true` 

If it checks if true, it will allow a person to turn it off without commenting out the setting or removing it completely. I think I would need `== true` and not just check if it's there. Maybe they're the same thing, I get confused with Ruby some times. It could be functioning exactly how I think it should. 

Also doublechecking that we're not checking `use_stealth_attacks` is set anymore. I kind of think that's what the setting in the spell is now for so there's no need, and the method used before `use_stealth_attacks?` checked the `use_stealth?` based on current exp so I think it's basically equivalent. 

Input needed.

**Comment by jayRyan24** [comment]
@rpherbig

File: combat-trainer.lic [hunk 1]
```diff
@@ -1200,6 +1200,8 @@ class SpellProcess
       @training_cyclic_timer = @training_cast_timer
     end
 
+    game_state.hide_on_cast = true if data['use_stealth']
```
Comment (rpherbig) [inline combat-trainer.lic]: 💡 Could this just as easily be `game_state.hide_on_cast = data['use_stealth']`?

**Comment by rpherbig** [comment]
Looks good to me. Do you feel like it's been tested enough?

File: combat-trainer.lic (hunk 1 above)
Comment (jayRyan24) [reply combat-trainer.lic]: Yeah, I'll change that later tonight.

**Comment by jayRyan24** [comment]
Sure, let me change it tonight and then tomorrow I'll get back to you. Thanks.

**Comment by jayRyan24** [comment]
Working for my Empath fine. I will test over night and also am trying to get necros involved.

**Comment by jayRyan24** [comment]
Working for a couple days without issue. Ready for merge.

**Comment by jayRyan24** [comment]
@rpherbig

**Pull request merged**